schemas/webpack.d.ts

```diff
@@ -0,0 +1,366 @@
+     */
+    devtool?: string | boolean;
+
+    /**
+     * The entry point(s) of the compilation.
+     *
+     * Object: Multiple entry bundles are created. The key is the chunk name. The value can be a string or an array. If its' array all modules are loaded upon startup. The last one is exported. If it's a string look at the definition for string.
+     * String: The entry point for one output file. The string is resolved to a module which is loaded upon startup.
+     */
+    entry: WebpackType.EntryItem | {[key: string]: WebpackType.EntryItem};
+
+    /**
+     * Specify dependencies that shouldn't be resolved by webpack, but should become dependencies of the resulting bundle. The kind of the dependency depends on `output.libraryTarget`.
+     *
+     * String: An exact matched dependency becomes external. The same string is used as external dependency.
+     * Object: If an dependency matches exactly a property of the object, the property value is used as dependency.
+     * Function: `function(context, request, callback(err, result))` The function is called on each dependency.
+     * RegExp: Every matched dependency becomes external.
+     * Array: Use the all externals options defined above.
+     */
+    externals?: WebpackType.Externals | WebpackType.Externals[];
+
+    /**
+     * Custom values available in the loader context.
+     */
+    loader?: WebpackType.Object;
+
+    /**
+     * Options affecting the normal modules (`NormalModuleFactory`).
+     */
+    module?: WebpackType.Module;
+
+    /**
+     * Name of the configuration. Used when loading multiple configurations.
+     */
+    name?: string;
+
+    /**
+     * Include polyfills or mocks for various node stuff.
+     */
+    node?: WebpackType.Node;
+
+    /**
+     * Options affecting the output of the compilation. `output` options tell webpack how to write the compiled files to disk.
+     */
+    output?: WebpackType.Output;
+
+    /**
+     * Add additional plugins to the compiler.
+     */
+    plugins?: WebpackType.Object[];
+
+    /**
+     * Capture timing information for each module.
+     */
+    profile?: boolean;
+
+    /**
+     * Store compiler state to a json file.
+     */
+    recordsInputPath?: string;
+
+    /**
+     * Load compiler state from a json file.
+     */
+    recordsOutputPath?: string;
+
+    /**
+     * Store/Load compiler state from/to a json file. This will result in persistent ids of modules and chunks. An absolute path is expected. `recordsPath` is used for `recordsInputPath` and `recordsOutputPath` if they left undefined.
+     */
+    recordsPath?: string;
+
+    /**
+     * Options affecting the resolving of modules.
+     */
+    resolve?: WebpackType.Resolve;
+
+    resolveLoader?: WebpackType.Resolve;
+
+    /**
+     * Used by the webpack CLI program to pass stats options.
+     */
+    stats?: WebpackType.Object | boolean | WebpackType.Stats;
+
+    /**
+     * TODO add description
```

Comment:
"Specifies webpack deployment target. This modifies how the webpack bootstrap function is generated based on each target. Can also expose extra global variables into the webpack module scope (required for specific targets (IE: `electron`))